src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -202,25 +197,79 @@ private static OtlpTrace.Span.Types.Event ToOtlpEvent(ActivityEvent activityEven
             return otlpEvent;
         }
 
+        private static OtlpCommon.AttributeKeyValue ToOtlpAttribute(KeyValuePair<string, string> kvp)
+        {
+            // TODO: enforce no duplicate keys?
+            // TODO: reverse?
+            // To maintain full fidelity to downstream receivers convert to the proper attribute types
+            var attrib = new OtlpCommon.AttributeKeyValue { Key = kvp.Key };
+
+            if (kvp.Value == null)

```

Comment:
OTel spec for attributes indicates that the key should be removed if the value is null:
> Attribute values of null are considered to be not set and get discarded as if that SetAttribute call had never been made. As an exception to this, if overwriting of values is supported, this results in clearing the previous value and dropping the attribute key from the set of attributes.

from https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes